vcd/resource_vcd_vm_insert_eject_media.go

```diff
@@ -0,0 +1,141 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/govcd"
+	"log"
+)
+
+func resourceVcdInsertEjectMedia() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdMediaInsert,
+		Delete: resourceVcdMediaEject,
+		Read:   resourceVcdVmInsertedMediaRead,
+
+		Schema: map[string]*schema.Schema{
+			"vdc": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				ForceNew: true,
+			},
+			"catalog": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "catalog name where to find media file",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "media name to use",
+			},
+			"vapp_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "vapp to use",
+			},
+			"vm_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "vm in vapp in which media will be inserted or ejected",
+			},
+		},
+	}
+}
+
+func resourceVcdMediaInsert(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] VM media insert initiated")
+
+	vm, org, err := getVM(d, meta)
+	if err != nil || org == (govcd.Org{}) {
+		return fmt.Errorf("error: %#v", err)
+	}
+
+	task, err := vm.HandleInsertMedia(&org, d.Get("catalog").(string), d.Get("name").(string))
+
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("error: %#v", err)
+	}
+
+	d.SetId(d.Get("vapp_name").(string) + "_" + d.Get("vm_name").(string) + "_" + d.Get("name").(string))
+	return resourceVcdVmInsertedMediaRead(d, meta)
+}
+
+func resourceVcdVmInsertedMediaRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[TRACE] VM insert media read initiated")
+
+	vm, _, err := getVM(d, meta)
+	if err != nil {
+		return fmt.Errorf("error: %#v", err)
+	}
+
+	isIsoMounted := false
+	for _, hardwareItem := range vm.VM.VirtualHardwareSection.Item {
+		if hardwareItem.ResourceSubType == "vmware.cdrom.iso" {
```

Comment:
This literal string is used several times in both go-vcloud-director and terraform prlugin.
It should be replaced by a constant in go-vcloud-director with a name that explains what it is.

Reply:
fixed